uniffi_bindgen/src/interface/mod.rs

```diff
@@ -1053,33 +1063,119 @@ impl APIConverter<Field> for weedle::dictionary::DictionaryMember<'_> {
         if let Type::Object(_) = type_ {
             bail!("Objects cannot currently appear in record fields");
         }
+        let default = match self.default {
+            None => None,
+            Some(v) => Some(v.value.convert(ci, &type_)?),
+        };
         Ok(Field {
             name: self.identifier.0.to_string(),
             type_,
             required: self.required.is_some(),
-            default: match self.default {
-                None => None,
-                Some(v) => Some(v.value.convert(ci)?),
-            },
+            default,
         })
     }
 }
 
+#[derive(Debug, Clone, Copy, Serialize, Deserialize, Hash)]
+pub enum Radix {
+    Decimal = 10,
+    Octal = 8,
+    Hexadecimal = 16,
+}
+
 // Represents a literal value.
 // Used for e.g. default argument values.
 #[derive(Debug, Clone, Serialize, Deserialize, Hash)]
 pub enum Literal {
     Boolean(bool),
     String(String),
-    // TODO: more types of literal
+    UInt(u64, Radix, Type),
+    Int(i64, Radix, Type),
+    Float(String, Type),
+    Enum(String, Type),
+    EmptySequence,
+    EmptyMap,
+    Null,
 }
 
-impl APIConverter<Literal> for weedle::literal::DefaultValue<'_> {
-    fn convert(&self, _ci: &ComponentInterface) -> Result<Literal> {
-        Ok(match self {
-            weedle::literal::DefaultValue::Boolean(b) => Literal::Boolean(b.0),
-            weedle::literal::DefaultValue::String(s) => Literal::String(s.0.to_string()),
-            _ => bail!("no support for {:?} literal yet", self),
+impl TypedAPIConverter<Literal> for weedle::literal::DefaultValue<'_> {
+    fn convert(&self, ci: &ComponentInterface, type_: &Type) -> Result<Literal> {
+        fn convert_integer(
+            literal: &weedle::literal::IntegerLit<'_>,
+            type_: &Type,
+        ) -> Result<Literal> {
+            use regex::Regex;
+
+            let (string, radix) = match literal {
+                weedle::literal::IntegerLit::Dec(v) => (v.0, Radix::Decimal),
+                weedle::literal::IntegerLit::Hex(v) => (v.0, Radix::Hexadecimal),
+                weedle::literal::IntegerLit::Oct(v) => (v.0, Radix::Octal),
+            };
+            // This is the radix of the parsed number, passed to `from_str_radix`.
+            let src_radix = radix as u32;
+            // This radix tells the backends how to represent the number in the output languages.
+            let dest_radix = if string == "0" || string.starts_with('-') {
+                // 1. weedle parses "0" as an octal literal, but we most likely want to treat this as a decimal.
```

Comment:
wow; octal zero I hope 😬

Reply:
:D Yeah. Thought that'd be a bit surprising to our API consumers finding `0o0` in the API.